content/blog/translating-rmd.md

```diff
@@ -0,0 +1,132 @@
+---
+author: "Rob Ashton"
+date: 2021-06-30
+title: Translating R Markdown
+best: false
+tags:
+ - R
+ - rmarkdown
+ - translation
+---
+
+## Translating Rmd
+
+As part of our work on [naomi](../../projects/naomi) we've been asked to produce reports from R which can be distributed to country teams across Sub-Saharan Africa in English, French and Portuguese. [R Markdown](https://rmarkdown.rstudio.com/docs/) has no prescribed method for translating documents, but there are many options available to make this possible. This post covers 3 possible approaches: code chunks, custom blocks and tabsets.
+
+## Code chunks
+
+The first method uses conditionally evaluated [code chunks](https://bookdown.org/yihui/rmarkdown/r-code.html). We can use R Markdown chunk options `include` and `eval` to hide output and stop a chunk from being evaluated. Combining with `results="asis"` this allows us to output text conditionally based on the `lang` parameter.
+
+````r
+---
+title: '&nbsp;'
+params:
+  lang: fr
+---
+
+```{r, include=FALSE}
+if (params$lang == "fr") {
+  print_fr <- TRUE
+  print_en <- FALSE
+} else {
+  print_fr <- FALSE
+  print_en <- TRUE
+}
+```
+
+
+```{r, echo=FALSE, results="asis", include=print_en, eval=print_en}
+cat("Plot of random points<br/>")
+plot(runif(10), runif(10), main = "Random points")
+```
+
+```{r, echo=FALSE, results="asis", include=print_fr, eval=print_fr}
+cat("Tracé de points aléatoires<br/>")
+plot(runif(10), runif(10), main = "Points aléatoires")
+```
+````
+
+This has the advantage that the chunk will only be evaluated for the output language. This is particularly important if any of the plots we're generating take a long time to compute. The downside is the text needs be wrapped in calls to `cat` and the formatting of the text has to be done manually. For example we need to add the line break at the end of the text or the plot and text will appear on the same line. As everything is written as a code chunk this means RStudio syntax highlighting for the text is not available.
```

Comment:
The duplication here looks like the biggest disadvantage to me, plus scaling this to (say) 5 languages would see very cumbersome